src/Microsoft.VisualStudio.SlowCheetah.VS/NugetHandler/PackageHandlers/BackgroundInstallationHandler.cs

```diff
@@ -47,18 +47,18 @@ public override void Execute(Project project)
             {
                 string warningTitle = this.IsUpdate ? Resources.Resources.NugetUpdate_Title : Resources.Resources.NugetInstall_Title;
                 string warningMessage = this.IsUpdate ? Resources.Resources.NugetUpdate_Text : Resources.Resources.NugetInstall_Text;
-                if (this.HasUserAcceptedWarningMessage(warningTitle, warningMessage))
+                if (await this.HasUserAcceptedWarningMessage(warningTitle, warningMessage))
                 {
                     // Gets the general output pane to inform user of installation
-                    IVsOutputWindowPane outputWindow = (IVsOutputWindowPane)this.Package.GetService(typeof(SVsGeneralOutputWindowPane));
+                    IVsOutputWindowPane outputWindow = (IVsOutputWindowPane)await this.Package.GetServiceAsync(typeof(SVsGeneralOutputWindowPane));
                     outputWindow?.OutputString(string.Format(CultureInfo.CurrentCulture, Resources.Resources.NugetInstall_InstallingOutput, project.Name) + Environment.NewLine);
 
-                    TPL.Task.Run(() =>
+                    var task = this.Package.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
This is slightly different than previous logic. `TPL.Task.Run` will post its supplied code block to the threadpool, so it always yields immediately. `JTF.RunAsync` however performs no posting of the sort. Really it only sets some JTF related stuff onto the execution context, and then invokes your code in the current context. So if this is the UI thread, the code below is UI thread as well. Essentially you have changed this from always running on the thread pool, to running wherever it is called from. You can solve this by immediately calling `await TaskScheduler.Default;` within the JTF.RunAsync.

Reply:
JTF also has a Task extension method `.Forget()`, it does nothing but it does make it clear you are intentionally fire & forgetting this task. Just so no one comes along later and is like "oh, someone forgot to await this task!" and adds the await.